File: src/main/java/dev/zprestige/fire/module/movement/noslow/ItemInputUpdateListener.java
package dev.zprestige.fire.module.movement.noslow;

import dev.zprestige.fire.event.bus.EventListener;
import dev.zprestige.fire.event.impl.ItemInputUpdateEvent;
import net.minecraft.util.MovementInput;

public class ItemInputUpdateListener extends EventListener<ItemInputUpdateEvent, NoSlow> {

    public ItemInputUpdateListener(final NoSlow noSlow) {
        super(ItemInputUpdateEvent.class, noSlow);
    }

    @Override
    public void invoke(final Object object) {
        if (module.slowed()) {
            final ItemInputUpdateEvent event = (ItemInputUpdateEvent) object;
            final MovementInput movementInput = event.getMovementInput();
            movementInput.moveForward /= 0.2f;
            movementInput.moveStrafe /= 0.2f;
        }
    }
}

File: src/main/java/dev/zprestige/fire/module/movement/noslow/ItemInputUpdateListener.kt
package dev.zprestige.fire.module.movement.noslow

import dev.zprestige.fire.event.bus.EventListener
import dev.zprestige.fire.event.impl.ItemInputUpdateEvent

class ItemInputUpdateListener(noSlow: NoSlow) : EventListener<ItemInputUpdateEvent, NoSlow>(ItemInputUpdateEvent::class.java, noSlow) {

    override fun invoke(e: Any) {
        if (module.slowed()) {
            val event = e as ItemInputUpdateEvent
            val movementInput = event.movementInput
            movementInput.moveForward /= 0.2f
            movementInput.moveStrafe /= 0.2f
        }
    }
}
